client/modules/Manage/Bookings/components/BookingAdm.js

```diff
@@ -0,0 +1,365 @@
+  }
+
+  handleBlur(field) {
+    this.setState({
+      isTouched: Object.assign({}, this.state.isTouched, { [field]: true }),
+    });
+  }
+
+  handleInputChange(evt)
+  {
+    let field = evt.target.id;
+    let value = evt.target.value
+    this.setState({
+      booking: { ...this.state.booking, [field]: value },
+      isTouched: { ...this.state.isTouched, [field]: true },
+    });
+  }
+
+  componentDidMount() {
+
+  }
+
+  renderTextFormGroup(field) {
+    const placeholder = this.labels[field];
+    return (
+      <FormGroup>
+        {this.renderLabel(field, field)}
+        <Input
+          type="text"
+          name={field}
+          id={field}
+          placeholder={placeholder}
+          className={this.isError(field) ? 'is-invalid' : ''}
+          onChange={this.handleInputChange.bind(this)}
+          onBlur={() => this.handleBlur(field)}
+          value={this.state.booking[field]}
+        />
+      </FormGroup>
+    );
+  }
+
+  toggleDropDown(field, evt) {
+    const dropdownState = {...this.state.dropdownsOpen};
+    dropdownState[field] = !dropdownState[field];
+    this.setState({dropdownsOpen: dropdownState});
+  }
+
+  ddOnSelect(field, value, evt) {
+    this.setState({
+      booking: {...this.state.booking, [field]: value},
+      isTouched: {...this.state.isTouched, [field]: true},
+      dropdownsOpen: {...this.state.dropdownsOpen, [field]: false},
+
+    });
+  }
+
+  renderDropdownFormGroup(field, options) {
+
+    const opts = options.map(m => (<DropdownItem key={m} onClick={(e) => this.ddOnSelect(field, m, e)}>{m}</DropdownItem>));
+    const text = this.state.booking[field];
+    return (
+      <FormGroup>
+          {this.renderLabel(field, field)} <br />
+          <ButtonDropdown isOpen={this.state.dropdownsOpen[field]} toggle={(e) => this.toggleDropDown(field, e)}>            
+            <DropdownToggle caret size="sm">{text}</DropdownToggle>
+            <DropdownMenu>
+              {opts}
+            </DropdownMenu>
+          </ButtonDropdown>          
+      </FormGroup>
+    );
+  }
+
+  disabledDropDown() {
+    const field = 'isDisabled';
+    const active = 'ACTIVE';
+    const disabled = 'INACTIVE';
+    const text = this.state.booking.isDisabled ? disabled : active;
+    
+    const opts = [
+      {state: false, text: active},
+      {state: true, text: disabled},
+    ].map(m => (<DropdownItem key={m.text} onClick={(e) => this.ddOnSelect(field, m.state, e)}>{m.text}</DropdownItem>));
+    
+    return (
+      <FormGroup>
+          {this.renderLabel(field, field)} <br />
+          <ButtonDropdown isOpen={this.state.dropdownsOpen[field]} toggle={(e) => this.toggleDropDown(field, e)}>            
+            <DropdownToggle caret size="sm">{text}</DropdownToggle>
+            <DropdownMenu>
+              {opts}
+            </DropdownMenu>
+          </ButtonDropdown>          
+      </FormGroup>
+    );
+  }
+
+  renderUserDropDown(field, options) {
```

Comment:
renderUSers and renderCars might be better off with a Typeahead component rather than a dropdown. that way an admin can search for a user or car if needed, whereas if there are hundreds of users or cars, a dropdown means reading and scrolling :)